migration-from-4x.md

```diff
@@ -0,0 +1,328 @@
+---
+layout: default
+title: Migrating from TinyMCE 4.x to TinyMCE 5.0.
+title_nav: Migrating from 4.x
+description: Instructions for migrating from TinyMCE 4.x to TinyMCE 5.0.
+keywords: migration considerations premigration pre-migration
+---
+
+## Migrating from 4.x
+
+The new TinyMCE 5.0 editor comes with significant changes to the previous versions, though most configuration changes affect complex use cases like adding custom components. Setting up a basic TinyMCE 5.0 instance should be no harder than with TinyMCE 4.x.
+
+Our team at Tiny has worked on creating a more configurable, more modern and less cumbersome editor while retaining the familiarity of the user interface from the older versions.
+
+This chapter describes the migration process and workarounds if you are using an older version of TinyMCE. It includes tasks that you must perform before the migration can begin, and different workaround procedures for deprecated, deleted, and updated features.
+
+## Editor-Core
+
+### Initialization
+
+The initialization process of TinyMCE 5.0 is the same as TinyMCE 4.x. The bootstrap process and initialization events all remain the same.  The main differences are in the `init` configuration, specifically the configuration items for UI components. It still retains a familiar JSON structure. However, many of the configuration options have been simplified.
+
+### Settings
+
+In TinyMCE 5.0, some configurations have been removed because they are no longer necessary or an improved solution has been introduced.
+
+#### Changed Settings
+
+| **Old Settings** |  **Description**|
+| ---------------- | --------------- |
+| Settings for inline styles | Uses CSS stylesheets for custom |
+
+TODO
+
+#### New Settings
+
+TODO
+
+#### Removed Settings
+
+* **fixed_toolbar_container**: Owing to the enhancements to the new inline toolbar behaviour, `fixed_toolbar_container` is not required in TinyMCE 5.0.
+
+TODO
+
+### Methods
+
+* All TinyMCE 4.x methods for creating creating UI components have been removed. New methods have been added for TinyMCE 5.0. For more information, see the [docs]({{site.baseurl}}/components/).
+* No core editor methods were removed (tinymce, editor, selection, on(), etc remain the same).
+
+## Themes
+
+In TinyMCE 5.0, some themes have been removed and are now combined in a new single responsive theme called **Silver**.
+
+| **Removed Theme** | **Replaced by** |
+| ----------------- | --------------  |
+| [Modern](https://www.tiny.cloud/docs/themes/modern/) | Silver |
+| [Modern inline](https://www.tiny.cloud/docs/general-configuration-guide/use-tinymce-inline/) | Silver Inline |
+| [Inlite(Distraction-free Editor)](https://www.tiny.cloud/docs/themes/inlite/) | Silver (distraction free configuration) |
+| [Mobile](https://www.tiny.cloud/docs/themes/mobile/) | Silver (responsive to screen size)  |
```

Comment:
@EkimChau Please confirm this table is still correct, re: mobile.

Reply:
@EkimChau confirmed that this information is accurate for TinyMCE 5.0.